### PR TITLE
[VarDumper] Prepare period caster for 4.0

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/23668
| License       | MIT
| Doc PR        | /

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 1]
```diff
@@ -20,6 +20,8 @@
  */
 class DateCaster
 {
+    private const PERIOD_LIMIT = 3;
+
```
Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: cf https://github.com/symfony/symfony/pull/23668#discussion_r135483156

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Component/VarDumper/Caster/DateCaster.php [hunk 2]
```diff
@@ -80,14 +82,10 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
 
     public static function castPeriod(\DatePeriod $p, array $a, Stub $stub, $isNested, $filter)
     {
-        if (defined('HHVM_VERSION_ID') || \PHP_VERSION_ID < 50620 || (\PHP_VERSION_ID >= 70000 && \PHP_VERSION_ID < 70005)) { // see https://bugs.php.net/bug.php?id=71635
-            return $a;
-        }
-
         $dates = array();
         if (\PHP_VERSION_ID >= 70107) { // see https://bugs.php.net/bug.php?id=74639
```
Comment (maidmaid) [inline src/Symfony/Component/VarDumper/Caster/DateCaster.php]: Minimal current PHP version is 7.1.3. What about bump it to 7.1.7 to avoid this condition?

Comment (nicolas-grekas) [reply src/Symfony/Component/VarDumper/Caster/DateCaster.php]: a caster is not enough a reason to bump the minimum version

**Comment by nicolas-grekas** [comment]
Thank you @maidmaid.

**Pull request merged**